### DISCUSSION TITLE
Is getInitialProps deprecated?

### DISCUSSION THREAD

**Discussion by no-name420** [vercel/next.js#17015] [Help] [answered]
Is getInitialProps deprecated? 
I'm asking because I see everywhere and in the official docs that Vercel is recommending to use getServerSideProps instead of getInitialProps, and it seems weird because there's still unique features that getInitialProps offers, for example there are cases where you don't mind if the code you run in getInitialProps is visible to the client and therefore there's no need to add the extra server call and use getServerSideProps...

**Comment by leerob** [comment, marked as answer]
No, `getInitialProps` is not deprecated. The reason the documentation recommends using `getStaticProps`/`getServerSideProps` is because allows you fall into the [pit of success](https://blog.codinghorror.com/falling-into-the-pit-of-success/).

Example: if you're only fetching data on the server-side, `getServerSideProps` helps you do the right thing by not including that code in the client-side bundle. Smaller bundle === faster page load times.

Hope this helps clarify!

**Reply by no-name420** [reply]
Thank you for the clarification 😀
